ci-scripts/_cluster-helpers.sh

```
@@ -0,0 +1,102 @@
+  if [[ -n "${OC_VERSION:-}" ]]; then
+    return
+  fi
+  OC_VERSION=$(oc version --output json 2>/dev/null \
+    | jq -r '.openshiftVersion | split(".") | .[0:2] | join(".") // empty') || true
+  OC_VERSION="${OC_VERSION:-4.20}"
+}
+
+# Resolve the internal image registry hostname from the cluster.
+# Sets INTERNAL_REGISTRY; defaults to the well-known service address if detection fails.
+resolve_internal_registry() {
+  INTERNAL_REGISTRY="$(oc get image.config.openshift.io/cluster \
```

Shellcheck (0.11.0)

[warning] INTERNAL_REGISTRY appears unused. Verify use (or export if used externally).

(SC2034)